libs/barista-components/button/src/_button-theme.scss

```
@@ -1,227 +1,271 @@
-@import '../../core/src/theming/theming';
-@import '../../core/src/style/interactive-common';
-
-@mixin dt-theme-button($theme) {
-  $palette-names: 'main', 'warning', 'cta';
-  $default-palette: 'main';
-
-  @each $name in $palette-names {
-    $is-dark: map-get($theme, 'is-dark');
-    $palette: dt-get-theme-palette($theme, $name);
-    $default-color: dt-get-theme-color($palette, 'default');
-    $hover-color: dt-get-theme-color($palette, 'hover');
-    $active-color: dt-get-theme-color($palette, 'active');
-
-    // Temporary fix to not make button color depend on theme.
-    // We only have to override the main color because the warning and
-    // cta colors are the same across all themes for now.
-    @if ($name == 'main') {
-      @if ($is-dark == true) {
-        $default-color: $turquoise-500;
-        $hover-color: $turquoise-400;
-        $active-color: $turquoise-300;
-      } @else {
-        $default-color: $turquoise-600;
-        $hover-color: $turquoise-700;
-        $active-color: $turquoise-800;
-      }
-    }
 
-    $namespace: '.dt-button';
-    @if $name != $default-palette {
-      $namespace: '.dt-button.dt-color-#{$name}';
-    }
+.dt-color-warning {
+  --dt-theme-main-default-color: var(--dt-warning-default-color);
+  --dt-theme-main-hover-color: var(--dt-warning-hover-color);
+  --dt-theme-main-active-color: var(--dt-warning-active-color);
```

I'm not entirely certain how good of an idea it is to reset the theme main color in this case. Maybe we should think about scoping the variable names to the button itself, this way we can prevent side effects if other components that allow nesting access i.e. the `--dt-theme-main-default-color`.

```suggestion
  --dt-button-default-color: var(--dt-warning-default-color);
  --dt-button-hover-color: var(--dt-warning-hover-color);
  --dt-button-active-color: var(--dt-warning-active-color);
```